### PR TITLE
Script for setting up thanos storage

### PR DESCRIPTION
**Pull request by ankushbbbr**
Python script to - 

1. Create S3 bucket
2. Create IAM Policy and attach it to user
3. Create k8s secret to be used by thanos

### CODE REVIEW

File: misc/setup_thanos_storage.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+                    "s3:PutObject",
+                    "s3:GetObject",
+                    "s3:AbortMultipartUpload",
+                    "s3:ListBucket",
+                    "s3:DeleteObject",
+                    "s3:ListMultipartUploadParts"
+                ],
+                "Resource": [
+                    "arn:aws:s3:::plantd-thanos",
+                    "arn:aws:s3:::plantd-thanos/*"
```
Comment (tomzhu1024) [inline misc/setup_thanos_storage.py]: The bucket name is hard-coded here. Should use the variable.

File: misc/setup_thanos_storage.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    # Create the IAM policy document
+    policy_document = {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Effect": "Allow",
+                "Action": [
+                    "s3:PutObject",
+                    "s3:GetObject",
+                    "s3:AbortMultipartUpload",
```
Comment (tomzhu1024) [inline misc/setup_thanos_storage.py]: Why are the permissions for multipart upload required?

File: misc/setup_thanos_storage.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    bucket_name = args.bucket_name
+    access_key = args.access_key
+    secret_key = args.secret_key
+
+    boto3.setup_default_session(
+        aws_access_key_id=access_key,
+        aws_secret_access_key=secret_key
+    )
+
+    namespace = "plantd-operator-system"
```
Comment (tomzhu1024) [inline misc/setup_thanos_storage.py]: Please put the magic constants to the front.

File: misc/setup_thanos_storage.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    identity = boto3.client('sts').get_caller_identity()
+    username = identity['Arn'].split('/')[-1]
+    account_id = identity['Account']
+    print(username, account_id)
+
+    create_s3_bucket(bucket_name)
+    policy_name = 'plantd-thanos-s3-1'
```
Comment (tomzhu1024) [inline misc/setup_thanos_storage.py]: Please put the magic constants to the front.

File: misc/setup_thanos_storage.py (hunk 1 above)
Comment (ankushbbbr) [reply misc/setup_thanos_storage.py]: fixed it

File: misc/setup_thanos_storage.py (hunk 3 above)
Comment (ankushbbbr) [reply misc/setup_thanos_storage.py]: done

File: misc/setup_thanos_storage.py (hunk 4 above)
Comment (ankushbbbr) [reply misc/setup_thanos_storage.py]: done

File: misc/setup_thanos_storage.py (hunk 2 above)
Comment (ankushbbbr) [reply misc/setup_thanos_storage.py]: right, those are not needed, removed it

**Review by tomzhu1024 — APPROVED** [review]

**Pull request merged**